### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by yuehhua**

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
+    "fourier",
+    "models",
+]
 
 @testset "NeuralOperators.jl" begin
-    include("preprocess.jl")
-    include("fourier.jl")
+    for f in tests
+        include("$f.jl")
+    end
```
Comment (foldfelis) [inline test/runtests.jl]: Such kind of declaration is not straight enough to me, and may decrease the readability.

File: test/fourier.jl [hunk 2]
```diff
@@ -27,22 +26,12 @@ end
...
-@testset "FNO" begin
-    𝐱, 𝐲 = get_data()
-    𝐱, 𝐲 = Float32.(𝐱), Float32.(𝐲)
-    @test size(FNO()(𝐱)) == size(𝐲)
-
-    m = FNO()
-    loss(𝐱, 𝐲) = sum(abs2, 𝐲 .- m(𝐱)) / size(𝐱)[end]
-    data = [(𝐱[:, :, 1:5], 𝐲[:, 1:5])]
-    Flux.train!(loss, params(m), data, Flux.ADAM())
+    @test true
```
Comment (foldfelis) [inline test/fourier.jl:36]: This is redundant

File: test/models.jl [hunk 3]
```diff
@@ -0,0 +1,11 @@
+@testset "FNO" begin
+    𝐱, 𝐲 = get_burgers_data()
+    𝐱, 𝐲 = Float32.(𝐱), Float32.(𝐲)
+    @test size(FNO()(𝐱)) == size(𝐲)
+
+    m = FNO()
+    loss(𝐱, 𝐲) = sum(abs2, 𝐲 .- m(𝐱)) / size(𝐱)[end]
+    data = [(𝐱[:, :, 1:5], 𝐲[:, 1:5])]
+    Flux.train!(loss, params(m), data, Flux.ADAM())
+    @test true
```
Comment (foldfelis) [inline test/models.jl:10]: This is redundant

File: test/fourier.jl [hunk 4]
```diff
@@ -9,13 +7,14 @@ using Flux
...
 
-    𝐱, _ = get_data()
+    𝐱, _ = get_burgers_data()
     @test size(m(𝐱)) == (64, 1024, 1000)
 
     T = Float32
     loss(x, y) = Flux.mse(real.(m(x)), y)
     data = [(T.(𝐱[:, :, 1:5]), rand(T, 64, 1024, 5))]
     Flux.train!(loss, params(m), data, Flux.ADAM())
+    @test true
```
Comment (foldfelis) [inline test/fourier.jl:17]: This is redundant

File: src/models.jl [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        Dense(2, 64, init=c_glorot_uniform),
+        FourierOperator(ch, modes, σ),
+        FourierOperator(ch, modes, σ),
+        FourierOperator(ch, modes, σ),
+        FourierOperator(ch, modes),
+        Dense(64, 128, σ, init=c_glorot_uniform),
+        Dense(128, 1, init=c_glorot_uniform),
+        flatten
+    )
+end
```
Comment (foldfelis) [inline src/models.jl]: Needs a new line to match the style

**Comment by foldfelis** [comment]
Thanks in advance

File: test/fourier.jl (hunk 2 above)
Comment (yuehhua) [reply test/fourier.jl:36]: Due to the use of `Flux.train`, it is a nondeterministic process. `@test true` is added to make sure the process is passed.

File: src/models.jl (hunk 5 above)
Comment (yuehhua) [reply src/models.jl]: Done.

File: test/runtests.jl (hunk 1 above)
Comment (yuehhua) [reply test/runtests.jl]: Done.

File: test/fourier.jl (hunk 2 above)
Comment (foldfelis) [reply test/fourier.jl:36]: The test case will still passed if the `train!` failed

**Review by foldfelis — COMMENTED** [review]
This PR can be safely closed due to #4 and #5 

Feel free to rebased if you got extra task to do.

**Pull request closed without merging**